Add episode 06 (Plotting).

mkcor opened the pull request:
Dear @maikia,

In the end, I couldn't help but give this introduction to plotting a scientific twist... :wink: 

I'll fix the conflicts in `./_episodes/CREDITS.md` after you merge #4.

I'm looping in @bmaz who kindly accepted to join as a helper!

Thank you,
Marianne

mkcor committed:
Add episode 06 (Plotting)

mkcor committed:
Add figures for episode 06

mkcor committed:
Document provenance

mkcor committed:
Add pandas dependency

maikia commented on _episodes/06-plotting.md:
```diff
@@ -0,0 +1,341 @@
+---
+title: "Plotting"
```
Previously you mentioned you like longer titles. How about this
```suggestion
title: "Visualize your data by plotting it"
```

mkcor replied:
Good point! But I'll keep the gerund form ("Visualizing ..."). Also, there's this thing about "data" being plural... :roll_eyes:

maikia commented on _episodes/06-plotting.md:
```diff
@@ -0,0 +1,341 @@
+- "How can I plot some data?"
+- "How can I save my plot for publishing?"
+objectives:
+- "Import a library into a program using `import libraryname`."
```
I am importing random in the previous lesson (no explanation given though, I thought I would just say it out loud)

mkcor replied:
Ok. If you don't mind, and since your episode is about functions, can you insist a little bit on the fact that libraries provide us with functions already written (by other people) that we can readily use (call)?

And, when I 'introduce' how to import a library, I'll echo your episode (it won't hurt). Thanks!

mkcor replied:
Hi @emilienschultz,

The above thread addresses the issue you opened at #15 :slightly_smiling_face:

maikia commented on _episodes/06-plotting.md:
```diff
@@ -0,0 +1,341 @@
+
+import pandas as pd
+
+df = pd.read_csv(url)
```
maybe you can also mention that it is possible to load the data from the computer using pandas

maikia replied:
maybe you should shortly add a sentence on what is a .csv format. And if they have a problem to load one of theirs properly to check 'delimiter' option.
and.. perhaps show them how to use help to understand what function is doing

mkcor replied:
Ok. Yes! I want to show how to get help when I introduce built-in functions, but I'll show it here again.

mkcor replied:
Done b449c5762f. I'll mention the 'delimiter' parameter only when speaking.

maikia commented on _episodes/06-plotting.md:
```diff
@@ -0,0 +1,341 @@
+What is this dataset about?
+
+~~~
+df[''description].unique()
```
```suggestion
df['description'].unique()
```

mkcor replied:
Thanks!

maikia reviewed (approved):
Welcome @bmaz in the team! :-D 

I am completely up for plotting! Although note that here you are also introducing loading the data from the file which is very important but perhaps it's introduced without enough explanation here?

I suppose we are leaving it just in case if there is time left, otherwise for them to do on their own? Do you want me to teach this one as well? I am happy to do so

mkcor commented on _episodes/CREDITS.md:
```diff
@@ -8,3 +8,5 @@
 [this episode](https://github.com/swcarpentry/python-novice-inflammation/blob/5c3ef8d/_episodes/04-lists.md).
 * Episode [Repeating Actions with Loops](./03-loops.md) is derived from
 [this episode](https://github.com/swcarpentry/python-novice-gapminder/blob/ace4b4f/_episodes/12-for-loops.md).
+* Episode [Plotting](./06-plotting.md) is derived from
```
```suggestion
* Episode [Visualizing data by plotting them](./06-plotting.md) is derived from
```

mkcor committed:
Apply suggestions from code review

Co-authored-by: Maria Telenczuk <telenczukm@gmail.com>

mkcor committed:
Link to episode 06 in lesson outline

mkcor committed:
Apply suggestions from code review

Co-authored-by: Maria Telenczuk <telenczukm@gmail.com>

mkcor committed:
Merge branch 'gh-pages' into add-episode-06

mkcor committed:
Add details about CSV files

mkcor commented:
@maikia I think I have addressed all of your comments. I've added a few bullet points in my personal instructor notes, so I don't forget to show and tell certain details.

maikia reviewed (approved):
LGTM thanks @mkcor
